code/modules/detective_work/scanner.dm

```diff
@@ -17,77 +17,85 @@
+	for(var/record in GLOB.data_core.general) // Search in the 'general' datacore
 		var/datum/data/record/S = record
-		if(S && (search == lowertext(S.fields["fingerprint"]) || search == lowertext(S.fields["name"])))
+		if(S && (search == lowertext(S.fields["fingerprint"]) || search == lowertext(S.fields["name"]))) // Get Fingerprint and Name
 			name = S.fields["name"]
 			fingerprint = S.fields["fingerprint"]
-			continue
+			break
 
-	for(var/record in GLOB.data_core.medical)
+	for(var/record in GLOB.data_core.medical) // Then search in the 'medical' datacore
 		var/datum/data/record/M = record
-		if(M && ( search == lowertext(M.fields["b_dna"]) || name == M.fields["name"]) )
+		if(M && (search == lowertext(M.fields["b_dna"]) || name == M.fields["name"])) // Get Blood DNA
 			dna = M.fields["b_dna"]
 
-			if(fingerprint == "FINGERPRINT NOT FOUND") // We have searched by DNA, and do not have the relevant information from the fingerprint records.
+			if(fingerprint == "FINGERPRINT NOT FOUND") // We have searched for DNA, and so do not have the relevant information from the fingerprint records.
 				name = M.fields["name"]
 				for(var/gen_record in GLOB.data_core.general)
 					var/datum/data/record/S = gen_record
 					if(S && (name == S.fields["name"]))
 						fingerprint = S.fields["fingerprint"]
-						continue
-			continue
+						break
 
 	if(name)
 		to_chat(user, "<span class='notice'>Match found in station records: <b>[name]</b></span><br>\
 		<i>Fingerprint:</i><span class='notice'> [fingerprint]</span><br>\
 		<i>Blood DNA:</i><span class='notice'> [dna]</span>")
-		return
-
-	to_chat(user, "<span class='warning'>No match found in station records.</span>")
+	else
+		to_chat(user, "<span class='warning'>No match found in station records.</span>")
 
 /obj/item/detective_scanner/ui_action_click()
 	print_scanner_report()
 
 /obj/item/detective_scanner/proc/print_scanner_report()
-	if(log.len && !scanning)
+	if(length(log) && !scanning)
 		scanning = 1
 		to_chat(usr, "<span class='notice'>Printing report, please wait...</span>")
 		playsound(loc, 'sound/goonstation/machines/printer_thermal.ogg', 50, 1)
-		spawn(100)
-
-			// Create our paper
-			var/obj/item/paper/P = new(get_turf(src))
-			P.name = "paper- 'Scanner Report'"
-			P.info = "<center><font size='6'><B>Scanner Report</B></font></center><HR><BR>"
-			P.info += jointext(log, "<BR>")
-			P.info += "<HR><B>Notes:</B><BR>"
-			P.info_links = P.info
-
-			if(ismob(loc))
-				var/mob/M = loc
-				M.put_in_hands(P)
-				to_chat(M, "<span class='notice'>Report printed. Log cleared.</span>")
-
-			// Clear the logs
-			log = list()
-			scanning = 0
+
+		addtimer(CALLBACK(src, .proc/make_paper, log), 10 SECONDS) // Create our paper
+		log = list() // Clear the logs
+		scanning = 0
 	else
 		to_chat(usr, "<span class='notice'>The scanner has no logs or is in use.</span>")
 
+/obj/item/detective_scanner/proc/make_paper(log) // Moved to a proc because 'spawn()' is evil
```

Comment:
Best to give this call an user. Tracing it back. `ui_action_click` gives an user and actiontype as param. The override in this file of that proc does not take those params but it should. That way you can pass the user

Reply:
Why would `user` be needed here? To replace the `if(ismob(loc))`?

If so, wouldn't that mean that only the user who printed the report gets in put into their hands?
(i.e. Detective prints report, drops the scanner, someone else picks it up.)

Reply:
Fair point there. Yeah in this case a loc check would make more sense due to the long delay